packages/workflow/src/Interfaces.ts

```diff
@@ -543,6 +544,15 @@ export interface IN8nRequestOperationPaginationOffset extends IN8nRequestOperati
 	};
 }
 
+export interface IN8NRequestOperationPaginationGeneric extends IN8nRequestOperationPaginationBase {
+	type: 'generic';
+	properties: {
+		continue: string;
+		request: IRequestOptionsSimplifiedAuth;
+		rootProperty?: string;
```

Comment:
This API is practically identical to what @janober originally suggested.

The only difference is I introduced the `rootProperty` attribute here separately as we cannot really make use of the postReceive action for that without refactoring it some. That is, if items are mapped between pages, we would need to be able to access the raw response to access e.g. the `nextCursor` property which is likely contained outside of the rootProperty.

Reply:
@krynble If I understood right, we would have something `postReceive` definable right inside the pagination options instead of the current `rootProperty`? That's not a bad idea at all! It would enable one to pull off any kinds of operations that would be needed (even if I would suspect extracting the `rootProperty` might be the most usual case).

If anything, it would be nice if we could somehow reuse the same code for handling postReceives as elsewhere, as not to create too many execution paths. I could give it a try and see how it feels.

Reply:
Yup that's the idea @mieky 

But I'm also thinking that we can remove the `rootProperty` from here and use the `postReceive` array from the top level.

If you check the `INodePropertyRouting` type, you'll see it contains an `output` attribute, which is an instance of `INodeRequestOutput`. This already contains the `postReceive?: PostReceiveAction[];` property.

So as part of `packages/workflow/src/RoutingNode.ts`, the `makeRoutingRequest` currently receives as argument `requestOperations?: IN8nRequestOperations,` which does not contain all the information we want. If instead we pass a higher level object (which would be `INodePropertyRouting`) we would have all the information we needed.

If this is unclear LMK and we can chat about it!